docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,608 @@
+### Populate existing objects
+
+The `JsonConvert.PopulateObject` method in `Newtonsoft.Json` deserializes a JSON document to an existing instance of a class, instead of creating a new instance. <xref:System.Text.Json> always creates a new instance of the target type by using the default public parameterless constructor. Custom converters can deserialize to an existing instance.
+
+### Reuse rather than replace properties
+
+The `Newtonsoft.Json` `ObjectCreationHandling` setting lets you specify that objects in properties should be reused rather than replaced during deserialization. <xref:System.Text.Json> always replaces objects in properties.  Custom converters can provide this functionality.
+
+### Add to collections without setters
+
+During deserialization, `Newtonsoft.Json` adds objects to a collection even if the property has no setter. <xref:System.Text.Json> ignores properties that don't have setters. Custom converters can provide this functionality.
+
+### MissingMemberHandling
+
+`Newtonsoft.Json` can be configured to throw exceptions during deserialization if the JSON includes properties that are missing in the target type. <xref:System.Text.Json> ignores extra properties in the JSON, except when you use the [[JsonExtensionData] attribute](system-text-json-how-to.md#handle-overflow-json). There's no workaround for the missing member feature.
+
+### TraceWriter
+
+`Newtonsoft.Json` lets you debug by using a `TraceWriter` to view logs that are generated by serialization or deserialization. <xref:System.Text.Json> doesn't do logging.
+
+## JsonDocument and JsonElement compared to JToken (like JObject, JArray)
+
+<xref:System.Text.Json.JsonDocument?displayProperty=fullName> provides the ability to parse and build a **read-only** Document Object Model (DOM) from existing JSON payloads. The DOM provides random access to data in a JSON payload. The JSON elements that compose the payload can be accessed via the <xref:System.Text.Json.JsonElement> type. The `JsonElement` type provides APIs to convert JSON text to common .NET types. `JsonDocument` exposes a <xref:System.Text.Json.JsonDocument.RootElement> property.
+
+### JsonDocument is IDisposable
+
+`JsonDocument` builds an in-memory view of the data into a pooled buffer. Therefore, unlike `JObject` or `JArray` from `Newtonsoft.Json`, the `JsonDocument` type implements `IDisposable` and needs to be used inside a using block. 
+
+Only return a `JsonDocument` from your API if you want to transfer lifetime ownership and dispose responsibility to the caller. In most scenarios, that isn't necessary. If the caller needs to work with the entire JSON document, return the <xref:System.Text.Json.JsonElement.Clone%2A> of the <xref:System.Text.Json.JsonDocument.RootElement%2A>, which is a <xref:System.Text.Json.JsonElement>. If the caller needs to work with a particular element within the JSON document, return the <xref:System.Text.Json.JsonElement.Clone%2A> of that <xref:System.Text.Json.JsonElement>.
```

Comment:
The way this is written is a lot clearer now :)

However, we should add some more info here for the other scenario. If the API accepts a `JsonElement` and is returning a subset of that `JsonElement`, then you don't need to call `Clone`. That is because the original lifetime owner of the JsonDocument which is what this `JsonElement` is attached to still owns the returned `JsonElement` as well anyway. Explaining that scenario would be good. The way it is written currently, folks might call `Clone` unnecessarily and that is expensive.

```C#
// This is perfectly fine to do because you are returning a subset of the input JsonElement
public JsonElement ReturnFileName(JsonElement source)
{
   return source.GetProperty("fileName");
}

// This is technically OK to do but calling Clone here is unnecessary and slow
public JsonElement ReturnFileName(JsonElement source)
{
   return source.GetProperty("fileName").Clone();
}
```